daemon/checkpoint.go

```diff
@@ -38,6 +38,11 @@ func (daemon *Daemon) CheckpointCreate(name string, config types.CheckpointCreat
 		return fmt.Errorf("Invalid checkpoint ID (%s), only %s are allowed", config.CheckpointID, validCheckpointNameChars)
 	}
 
+	checkDir := filepath.Join(daemon.repository, container.ID, "checkpoints", config.CheckpointID)
+	if stat, err := os.Stat(checkDir); err == nil && stat.IsDir() {
+		return fmt.Errorf(" checkpoint with name %s already exists for container %s", config.CheckpointID, container.Name)
+	}
+
 	err = daemon.containerd.CreateCheckpoint(container.ID, config.CheckpointID, checkpointDir, config.Exit)
 	if err != nil {
```

Comment:
@dattatrayakumbhar yes, I think that's something that needs to be improved, or we should remove the functionality of a custom checkpoint-dir (and replace with, e.g., an option so save the checkpoint data to a tar, and to load from that tar).

Checkpoint/restore is still experimental, and UX needs feedback (and improvement)